core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1238,6 +1239,18 @@ package object config {
         s"The value must be in allowed range [1,048,576, ${MAX_BUFFER_SIZE_BYTES}].")
       .createWithDefault(1024 * 1024)
 
+  private[spark] val UNSAFE_SORTER_SPILL_READER_BUFFER_SIZE_RATIO =
+    ConfigBuilder("spark.unsafe.sorter.spill.reader.buffer.size.ratio")
```

Comment:
Just using size for this param is a bad idea? Why do you use ratio instead of size?

Reply:
Perhaps it is my misunderstanding of your comment from 14 days ago.
You mention following:

private byte[] arr = new byte[1024 *** 1024**];
If this number is performance-senstive, could we parameterize it?

So, I created parameter. Is my understanding incorrect?

Why do you use ratio instead of size?
There is existing parameter “spark.unsafe.sorter.spill.reader.buffer.size” which is used for different purpose, so I am not able to use it here. Also, I followed your recommendation to parameterize ratio. I am OK with creating parameter for entire buffer size instead ratio. 
Please let me know what makes more sense to you.

Reply:
I originally meant;
```
// package.scala
  private[spark] val UNSAFE_SORTER_SPILL_READER_BUFFER_DEFAULT_SIZE_BYTES =
    ConfigBuilder("spark.unsafe.sorter.spill.reader.buffer.defaultSize")
      .internal()
      .bytesConf(ByteUnit.BYTE)
      .checkValue(...)
      .createWithDefault(1024 * 1024);

// UnsafeSorterSpillReader.java 
  private byte[] arr = null;

  public UnsafeSorterSpillReader(
      SerializerManager serializerManager,
      File file,
      BlockId blockId) throws IOException {
    assert (file.length() > 0);
    int defaultBufferSize = (int)SparkEnv.get().conf().get(package$.MODULE$.UNSAFE_SORTER_SPILL_READER_BUFFER_SIZE_BYTES())
    this.arr = new byte[DEFAULT_BUFFER_SIZE_RATIO];
```
Am I miss something?

Reply:
Thank you for the clarification. It was my misunderstanding.